spark/v3.4/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/optimizer/FoldConstantSystemFunction.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.spark.sql.catalyst.optimizer
+
+import org.apache.iceberg.spark.functions.UnboundFunctionWrapper.SystemFunctionWrapper
+import org.apache.spark.sql.catalyst.expressions.Alias
+import org.apache.spark.sql.catalyst.expressions.ApplyFunctionExpression
+import org.apache.spark.sql.catalyst.expressions.EmptyRow
+import org.apache.spark.sql.catalyst.expressions.Literal
+import org.apache.spark.sql.catalyst.plans.logical.LocalRelation
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.catalyst.plans.logical.Project
+import org.apache.spark.sql.catalyst.rules.Rule
+
+/**
+ * Folding those system function that applies a constant value.
+ */
+object FoldConstantSystemFunction extends Rule[LogicalPlan] {
+
+  override def apply(plan: LogicalPlan): LogicalPlan = plan transform {
+    case logicalPlan: LogicalPlan =>
+      logicalPlan transformExpressions {
+        case applyFunc: ApplyFunctionExpression
+             if applyFunc.function.isInstanceOf[SystemFunctionWrapper[_]] =>
+          val fakeProjectList = applyFunc.children.zipWithIndex.map(pair => Alias(pair._1, s"fake_${pair._2}")())
```

Comment:
Why is the alias needed?

Reply:
The project list of `Project` should be `NamedExpression`